source/file/file.go

```diff
@@ -1,31 +1,24 @@
 package file
 
 import (
-	"net/http"
 	nurl "net/url"
 	"os"
 	"path/filepath"
 
 	"github.com/golang-migrate/migrate/v4/source"
-	"github.com/golang-migrate/migrate/v4/source/httpfs"
 )
 
 func init() {
 	source.Register("file", &File{})
 }
 
-type File struct {
-	httpfs.PartialDriver
-	url  string
-	path string
-}
+type File = file
```

Comment:
Is a type alias necessary? I think you can define `type struct File` in both the go 1.16 and non-1.16 files.

Reply:
What kind of struct do you envision?
Sorry, I can't understand well.

Reply:
I *think* you can avoid the type alias by doing:
```golang
// In file_go115.go
type File struct {
	httpfs.PartialDriver
	url  string
	path string
}

// In file_go116.go
type File struct {
	iofs.PartialDriver
	url  string
	path string
}
```